classes/class-wc-connect-account-settings.php

```diff
@@ -29,7 +29,7 @@ public function get() {
 		}
 
 		$master_user    = WC_Connect_Jetpack::get_master_user();
-		$connected_data = WC_Connect_Jetpack::get_connected_user_data( $master_user->ID );
+		$connected_data = $master_user ? WC_Connect_Jetpack::get_connected_user_data( $master_user->ID ) : false;
```

Comment:
Is this to handle jetpack not being connected? Since WCS requires jetpack, should we return false on the entire function? Or return payment information + blanking out user detail? I don't think there would be any payment methods if it's not connected.

Reply:
When I used [this file](https://github.com/Automattic/woocommerce-services/blob/dc6edb776bae802dfa56259d9a6cd1c2c12361c6/tests/e2e/config/travis/wc-services-testing-helper.php) to mock out Jetpack, I discovered that `WC_Connect_Jetpack::get_master_user()` would start returning either undefined or an empty array/object--I can't remember which--but basically this meant that any reference to its member variables (e.g. `$master_user->ID`, `$master_user->display_name`) threw a PHP error notice on the WCS settings page stating the index was undefined. I wasn't entirely sure of all the places where this function is used, so I just handled the error and returned empty strings for those variables, but @harriswong let me know if you think it would be better to just return `false` for the entire function here in that case.

Reply:
Is there a way to get `WC_Connect_Jetpack::get_master_user()` to return a mock master user when we are testing? 

> let me know if you think it would be better to just return false for the entire function here in that case.

Here are my thoughts. This change seems to be here because we need it to get E2E test to work. I generally prefer not to modify source code because of test; if possible, I would prefer to modify/hack the tests instead. The change itself will not prevent an error to be thrown, would that change a 500 to a 200? Would this allow account settings to be displayed/returned when it shouldn't? If the latter are true, then maybe there are some risks exposing data. That said, WCS requires jetpack to connect; without Jetpack, this code should not run so the change here shouldn't affect anything.

If we can modify test code instead of this, it would be great; otherwise, I think this is okay.

Reply:
> I generally prefer not to modify source code because of test; if possible, I would prefer to modify/hack the tests instead.

Totally agree with this sentiment. I had a bit of a play around today, but I managed to add a few lines to [`wc-services-testing-helper.php`](https://github.com/Automattic/woocommerce-services/blob/add/e2e-docker-container/tests/e2e/config/travis/wc-services-testing-helper.php#L27) to allow me to mock out these fields and revert this file back to the way it was before!